internal/transactionsubmission/engine/signing/channel_account_db_signature_client.go

```diff
@@ -0,0 +1,212 @@
+		ledgerNumberTracker:  opts.LedgerNumberTracker,
+	}, nil
+}
+
+var _ SignatureClient = &ChannelAccountDBSignatureClient{}
+
+func (c *ChannelAccountDBSignatureClient) getKPsForAccounts(ctx context.Context, stellarAccounts ...string) ([]*keypair.Full, error) {
+	if len(stellarAccounts) == 0 {
+		return nil, fmt.Errorf("no accounts provided")
+	}
+
+	accountsAlreadyAccountedFor := map[string]struct{}{}
+	kps := []*keypair.Full{}
+	for i, account := range stellarAccounts {
+		if _, ok := accountsAlreadyAccountedFor[account]; ok {
+			continue
+		}
+		accountsAlreadyAccountedFor[account] = struct{}{}
+
+		if account == "" {
+			return nil, fmt.Errorf("account %d is empty", i)
+		}
+
+		// Can return ErrRecordNotFound
+		chAcc, err := c.chAccModel.Get(ctx, c.dbConnectionPool, account, 0)
+		if err != nil {
+			return nil, fmt.Errorf("getting secret for channel account %q: %w", account, err)
+		}
+
+		chAccPrivateKey, err := c.encrypter.Decrypt(chAcc.PrivateKey, c.encryptionPassphrase)
+		if err != nil {
+			return nil, fmt.Errorf("cannot decrypt private key: %w", err)
+		}
+
+		kp, err := keypair.ParseFull(chAccPrivateKey)
+		if err != nil {
+			return nil, fmt.Errorf("parsing secret for channel account %q: %w", account, err)
+		}
+		kps = append(kps, kp)
+	}
+
+	return kps, nil
+}
+
+func (c *ChannelAccountDBSignatureClient) SignStellarTransaction(ctx context.Context, stellarTx *txnbuild.Transaction, stellarAccounts ...string) (signedStellarTx *txnbuild.Transaction, err error) {
+	if stellarTx == nil {
+		return nil, fmt.Errorf("stellarTx cannot be nil in %s", c.Type())
+	}
+
+	kps, err := c.getKPsForAccounts(ctx, stellarAccounts...)
+	if err != nil {
+		return nil, fmt.Errorf("getting keypairs for accounts %v in %s: %w", stellarAccounts, c.Type(), err)
+	}
+
+	signedStellarTx, err = stellarTx.Sign(c.NetworkPassphrase(), kps...)
+	if err != nil {
+		return nil, fmt.Errorf("signing transaction in %s: %w", c.Type(), err)
+	}
+
+	return signedStellarTx, nil
+}
+
+func (c *ChannelAccountDBSignatureClient) SignFeeBumpStellarTransaction(ctx context.Context, feeBumpStellarTx *txnbuild.FeeBumpTransaction, stellarAccounts ...string) (signedFeeBumpStellarTx *txnbuild.FeeBumpTransaction, err error) {
+	if feeBumpStellarTx == nil {
+		return nil, fmt.Errorf("stellarTx cannot be nil in %s", c.Type())
+	}
+
+	kps, err := c.getKPsForAccounts(ctx, stellarAccounts...)
+	if err != nil {
+		return nil, fmt.Errorf("getting keypairs for accounts %v in %s: %w", stellarAccounts, c.Type(), err)
+	}
+
+	signedFeeBumpStellarTx, err = feeBumpStellarTx.Sign(c.NetworkPassphrase(), kps...)
+	if err != nil {
+		return nil, fmt.Errorf("signing transaction in %s: %w", c.Type(), err)
+	}
+
+	return signedFeeBumpStellarTx, nil
+}
+
+func (c *ChannelAccountDBSignatureClient) BatchInsert(ctx context.Context, amount int) (publicKeys []string, err error) {
+	if amount < 1 {
+		return nil, fmt.Errorf("the amnount of accounts to insert need to be greater than zero")
```

Comment:
```suggestion
		return nil, fmt.Errorf("the amount of accounts to insert need to be greater than zero")
```
nit: I would use number instead of amount here. (amount for mass nouns, number for count nouns)

Reply:
FIxed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/174/commits/7a02ff18489ce2aa017b64f46bee7703f7f3e27d.